### DISCUSSION TITLE
Strategies for passing files between edge peers

### DISCUSSION THREAD

**Discussion by christroutner** [libp2p/js-libp2p#2476] [Q&A]
I'm using js-libp2p and helia in a node.js environment. I don't have a lot of browser-based nodes, which is what webRTC seems to focus on.

My p2p network has a lot of IPFS nodes running in Docker containers on desktops behind multiple firewalls. A good example of this is a VM like VirtualBox running Ubuntu on a desktop PC. There is the the VM firewall, then the host firewall, then the router firewall, all before a connection can reach the internet.

I have bootstrap nodes set up on cloud VPSs. They have a public IP4 address and open ports. So the edge nodes can all connect to them and communicate over pubsub channels. 

But they can not pass files to one another through these boostrap nodes, and I need this capability. The only way to pass files seems to be if they can connect directly to one another.

What I need help with is coming up with a strategy for edge nodes to connect to one another and be able to pass files to one another. V1 Circuit Relays were a great option for this. However I have not been able to get nodes to connect to one another over webRTC or V2 circuit relays.

Am I describing a situation where webRTC and v2 Circuit Relays are expected to work? Because that hasn't been my experience. 

Is there another strategy that can be employed to help these edge nodes connect in a way to be able to pass files to one another?

**Comment by master255** [comment]
The best strategy for transferring files between two peers has always been and always will be torrents. Use torrents!
